builds/e2e/templates/deploy-iotuap-setup.yaml

```diff
@@ -0,0 +1,49 @@
+parameters:
+  edgelet.artifact.name: ''
+  images.artifact.name: ''
+
+steps:
+  - task: CopyFiles@2
+    displayName: 'Copy Edgelet Artifact'
+    inputs:
+      SourceFolder: "$(Build.StagingDirectory)/${{ parameters['edgelet.artifact.name'] }}"
+      TargetFolder: "\\\\$(iotuap-x64-machine)\\c$\\Data\\e2etests\\artifacts\\${{ parameters['edgelet.artifact.name'] }}"
+      CleanTargetFolder: true
+  - task: CopyFiles@2
+    displayName: 'Copy Images Artifact'
+    inputs:
+      SourceFolder: "$(Build.StagingDirectory)/${{ parameters['images.artifact.name'] }}"
+      TargetFolder: "\\\\$(iotuap-x64-machine)\\c$\\Data\\e2etests\\artifacts\\${{ parameters['images.artifact.name'] }}"
+      CleanTargetFolder: true
+  - task: PowerShellOnTargetMachines@3
+    displayName: 'Update iotedged'
+    inputs:
+      Machines: '$(iotuap-x64-machine)'
+      UserName: '$(iotuap-x64-login)'
+      UserPassword: '$(iotuap-x64-password)'
+      InlineScript: |
+        # Check iotedge version
+        $serviceName = 'iotedge'
+        
+        If (Get-Service $serviceName -ErrorAction SilentlyContinue) {
+            $serviceVersion = Get-Service $serviceName version
+            $serviceVersion.Substring($serviceVersion.IndexOf("dev")+3,$serviceVersion.IndexOf(" (")-$serviceVersion.IndexOf("dev")-3)
+            If ($(RELEASE.ARTIFACTS.EDGELET.BUILDNUMBER) -gt $serviceVersion) {
+                If ((Get-Service $serviceName).Status -eq 'Running') {
+                    Stop-Service $serviceName
+                }
+                Write-Host "Updating $serviceName..."
+                Update-IoTEdge -ContainerOs Windows -OfflineInstallationPath C:\Data\e2etests\artifacts\iotedged-windows
+            } Else {
+                Write-Host "$serviceName is up to date."
+                # shutdown -r
+            }
+        } Else {
+            Write-Host "$serviceName not found."
+            Write-Host "Installing $serviceName..."
+            . C:\Data\e2etests\artifacts\core-windows\scripts\windows\setup\IotEdgeSecurityDaemon.ps1
+            deploy-iotedge
+        }
+      CommunicationProtocol: Http
+      ErrorActionPreference: silentlyContinue
+    condition: succeededOrFailed()
```

Comment:
does this means this step will be run no matter previous steps are succeeded or failed? if yes, remove it. only run if all previous tasks are passed.

Reply:
Removed.